src/components/Canvas.vue

```diff
@@ -63,22 +76,67 @@ export default {
   },
   methods: {
     onConnect (event) {
-      this.newEdge = {
-        start: event.connector,
-        end: {
-          x: event.position.left - this.$el.offsetLeft,
-          y: event.position.top - this.$el.offsetTop
-        },
-        color: types.colors[event.instance.spec.type]
-      }
-      if (!event.isOutput) this.newEdge.start = [this.newEdge.end, this.newEdge.end = this.newEdge.start][0]
       if (event.isFinal) {
+        if (this.newEdge && event.connector !== this.newEdge.start) {
+          const edge = {
+            start: this.newEdge.start,
+            end: event.instance,
+            color: this.newEdge.color
+          }
+          if (event.isOutput) edge.start = [edge.end, edge.end = edge.start][0]
+          this.addEdge(edge)
+        }
         this.newEdge = null
         event.instance.connecting = false
-      } else event.instance.connecting = true
+      } else {
+        this.newEdge = {
+          start: event.instance,
+          end: {
+            x: event.position.x - this.$el.offsetLeft,
+            y: event.position.y - this.$el.offsetTop
+          },
+          color: types.colors[event.instance.spec.type]
+        }
+        if (!event.isOutput) this.newEdge.start = [this.newEdge.end, this.newEdge.end = this.newEdge.start][0]
+        event.instance.connecting = true
+        if (this.$refs.newEdge) this.$refs.newEdge.refresh()
+      }
     },
-    addNode: function (component, spec) {
+    onNodeMove (nodeId) {
+      for (let edgeId in this.$refs) {
+        if (edgeId.split('.').includes(nodeId)) {
+          for (let edge of this.$refs[edgeId]) {
+            edge.refresh()
```

Comment:
Ich wein gleich

Reply:
Should this perhaps also refresh newEdge? Vinodh's multi-touch might lead to visual artifacts, right?

Reply:
Very good point